docs/DafnyRef/UserGuide.md

```diff
@@ -319,17 +319,45 @@ This list is not exhaustive but can definitely be useful to provide the next ste
 To understand how to control verification,
 it is first useful to understand how Dafny verifies functions and methods.
 
-For every method (or function, constructor, etc.), Dafny extracts _assertions_, for example:
+For every method (or function, constructor, etc.), Dafny extracts _assertions_. Here is a non-exhaustive list of such extracted assertions:
 
-* any explicit [`assert` statement](#sec-assert-statement) is _an assertion_[^precision-requires-clause].
-* A consecutive pair of lines in a [`calc` statement](#sec-calc-statement) forms _an assertion_ that the expressions are related according to the common operator.
-* Every call to a function or method with a [`requires` clause](#sec-requires-clause) yields _one assertion per requires clause_[^precision-requires-clause]
-  (special cases such as sequence indexing come with a special require clause that the index is within bounds).
-* Assignments `o.f := E;` yield an _assertion_ that `o.f` is allowed by the enclosing [`modifies` clause](#sec-loop-framing).
-* [Assign-such-that operators](#sec-update-and-call-statement) `x :| P(x)` yield an _assertion_ that `exists x :: P(x)`.
-* Every [`ensures` clause](#sec-ensures-clause) yields an _assertion_ at the end of the method and on every return.
-* Every [`decreases` clause](#sec-decreases-clause) yields an _assertion_ at either a call site or at the end of a while loop.
+**Integer assertions:**
+
+* Every [division](#sec-numeric-types) yields an _assertion_ that the divisor is never zero.
+* Every [bounded number operation](#sec-numeric-types) yields an _assertion_ that the result will have the same bounds (no overflow, no underflows).
+* Every [conversion](#sec-as-expression) yields an _assertion_ that conversion is compatible
+* Every [bitvector shift](#sec-bit-vector-types) yields an _assertion_ that the shift amount is never negative, and that the shift amount is within the width of the value.
+
+**Object assertions:**
+
+* Every [object property access](#sec-class-types) yields an _assertion_ that the object is not null
+* Every assignment `o.f := E;` yield an _assertion_ that `o` is among the set of objects of the enclosing [`modifies`](#sec-loop-framing) clause in loops, or the enclosing [`modifies`](#sec-modifies-clause) clause on methods.
+* Every read `o.f` yield an _assertion_ that `o` is among the set of objects of the enclosing [`reads`](#sec-reads-clause) clause on functions and predicates, or the enclosing [`modifies`](#sec-modifies-clause) clause on methods
```

Comment:
```suggestion
* Every read `o.f` yield an _assertion_ that `o` is among the set of objects of the [`reads`](#sec-reads-clause) clause of the function, predicate, or method; or the enclosing [`modifies`](#sec-modifies-clause) clause of the method.
```

Reply:
I'm not 100% sure that my suggested change is true. We should investigate.

Reply:
There is no reads clause on methods (yet)

Reply:
I modified your comment slightly to commit it.

Reply:
Sounds great.